cloudinary-core/src/main/java/com/cloudinary/metadata/MetadataValidation.java

```diff
@@ -0,0 +1,177 @@
+     * @param <T>
+     */
+    abstract static class ComparisonRule<T> extends MetadataValidation {
+        ComparisonRule(String type, T value) {
+            this(type, value, null);
+        }
+
+        ComparisonRule(String type, T value, Boolean equals) {
+            put(TYPE, type);
+            putValue(value);
+            if (equals != null) {
+                put(EQUALS, equals);
+            }
+        }
+
+        protected void putValue(T value) {
+            put(VALUE, value);
+        }
+    }
+
+    /**
+     * Great-than rule for integers.
+     */
+    public static class IntGreaterThan extends ComparisonRule<Integer> {
+        /**
+         * Create a new rule with the given integer.
+         * @param value The integer to reference in the rule
+         */
+        public IntGreaterThan(Integer value) {
+            super(GREATER_THAN, value);
+        }
+
+        /**
+         * Create a new rule with the given integer.
+         * @param value The integer to reference in the rule.
+         * @param equals Whether a field value equal to the rule value is considered valid.
+         */
+        public IntGreaterThan(Integer value, Boolean equals) {
+            super(GREATER_THAN, value, equals);
+        }
+    }
+
+    /**
+     * Great-than rule for dates.
+     */
+    public static class DateGreaterThan extends ComparisonRule<Date> {
+        /**
+         * Create a new rule with the given date.
+         * @param value The integer to reference in the rule
+         */
+        public DateGreaterThan(Date value) {
+            super(GREATER_THAN, value);
+        }
+
+        /**
+         * Create a new rule with the given date.
+         * @param value The date to reference in the rule.
+         * @param equals Whether a field value equal to the rule value is considered valid.
+         */
+        public DateGreaterThan(Date value, Boolean equals) {
+            super(GREATER_THAN, value, equals);
+        }
+
+        @Override
+        protected void putValue(Date value) {
+            put(VALUE, ObjectUtils.toISO8601DateOnly(value));
+        }
+    }
+
+    /**
+     * Less-than rule for integers.
+     */
+    public static class IntLessThan extends ComparisonRule<Integer> {
+        /**
+         * Create a new rule with the given integer.
+         * @param value The integer to reference in the rule
+         */
+        public IntLessThan(Integer value) {
+            super(LESS_THAN, value);
+        }
+
+        /**
+         * Create a new rule with the given integer.
+         * @param value The integer to reference in the rule.
+         * @param equals Whether a field value equal to the rule value is considered valid.
+         */
+        public IntLessThan(Integer value, Boolean equals) {
+            super(LESS_THAN, value, equals);
+        }
+    }
+
+    /**
+     * Less-than rule for dates.
+     */
+    public static class DateLessThan extends ComparisonRule<Date> {
```

Comment:
Really? that's the way we want to handle every rule that will be added? 
when the BE will support dateBetween, we will need to extend this class? 
Why not do something more generic that gets a type and an operator?

Reply:
That's actually already supported. We just need to make it public instead of abstract to let our users touch that, if you think it's simpler for the users. The derived classes are basically just helpers and can be removed if you want to. This code already works:
`new ComparisonRule<Date>(GREATER_THAN, new Date()){}`

Also, dateBetween is already supported (GT min AND LT max as per product spec).

Reply:
new ComparisonRule<Date>(GREATER_THAN, new Date()){}
makes much more sense